ena-submission/scripts/ena_submission_helper.py

```diff
@@ -144,10 +159,266 @@ def get_project_xml(project_set):
+
+
 def post_webin(xml, config: ENAConfig):
     return requests.post(
         config.ena_submission_url,
         auth=HTTPBasicAuth(config.ena_submission_username, config.ena_submission_password),
         files=xml,
         timeout=10,  # wait a full 10 seconds for a response incase slow
     )
+
+
+def create_chromosome_list(list_object: AssemblyChromosomeListFile) -> str:
+    """
+    Creates a temp file chromosome list:
+    https://ena-docs.readthedocs.io/en/latest/submit/fileprep/assembly.html#chromosome-list-file
+    """
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".gz") as temp:
+        filename = temp.name
+
+    with gzip.GzipFile(filename, "wb") as gz:
+        for entry in list_object.chromosomes:
+            gz.write(
+                f"{entry.object_name}\t{entry.chromosome_name}\t{entry.topology!s}-{entry.chromosome_type!s}\n".encode()
+            )
+
+    return filename
+
+
+def create_fasta(
+    unaligned_sequences: dict[str, str], chromosome_list: AssemblyChromosomeListFile
+) -> str:
+    """
+    Creates a temp fasta file:
+    https://ena-docs.readthedocs.io/en/latest/submit/fileprep/assembly.html#fasta-file
+    """
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".fasta.gz") as temp:
+        filename = temp.name
+
+    with gzip.GzipFile(filename, "wb") as gz:
+        if len(unaligned_sequences.keys()) == 1:
+            entry = chromosome_list.chromosomes[0]
+            gz.write(f">{entry.object_name}\n".encode())
+            gz.write(f"{unaligned_sequences["main"].replace('-', 'N')}\n".encode())
+        else:
+            for entry in chromosome_list.chromosomes:
+                gz.write(f">{entry.object_name}\n".encode())
+                gz.write(
+                    f"{unaligned_sequences[entry.chromosome_name].replace('-', 'N')}\n".encode()
+                )
+
+    return filename
+
+
+def create_manifest(manifest: AssemblyManifest):
+    """
+    Creates a temp manifest file:
+    https://ena-docs.readthedocs.io/en/latest/submit/assembly/genome.html#manifest-files
+    """
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".tsv") as temp:
+        filename = temp.name
+    with open(filename, "w") as f:
+        f.write(f"STUDY\t{manifest.study}\n")
+        f.write(f"SAMPLE\t{manifest.sample}\n")
+        f.write(
+            f"ASSEMBLYNAME\t{manifest.assemblyname}\n"
+        )  # This is the alias that needs to be unique
+        f.write(f"ASSEMBLY_TYPE\t{manifest.assembly_type!s}\n")
+        f.write(f"COVERAGE\t{manifest.coverage}\n")
+        f.write(f"PROGRAM\t{manifest.program}\n")
+        f.write(f"PLATFORM\t{manifest.platform}\n")
+        f.write(f"FASTA\t{manifest.fasta}\n")
+        f.write(f"CHROMOSOME_LIST\t{manifest.chromosome_list}\n")
+        if manifest.description:
+            f.write(f"DESCRIPTION\t{manifest.description}\n")
+        if manifest.moleculetype:
+            f.write(f"MOLECULETYPE\t{manifest.moleculetype!s}\n")
```

Comment:
Is this safe? It looks like a potential source for an injection. Should we perform some escaping or validations (e.g., check that user-specified fields such as platform/sequencingProtocol do not contain forbidden/problematic characters)?

Reply:
I am not quite sure what you mean - this function produces a manifest file and isn't added to the postgres database. The file is sent to ENA using their webin-cli so I don't think there is any injection risk here? But maybe I don't understand?

Reply:
created a follow up issue: https://github.com/loculus-project/loculus/issues/2831